From 7207fd3bf889b5ef2cf4fc27c685d3658f3309a9 Mon Sep 17 00:00:00 2001
From: Fred Chen <Fred.Chen@quantatw.com>
Date: Fri, 7 Aug 2026 17:49:42 +0800
Subject: [PATCH 10/11] configutations: santabarbara: add pdb1 3rd source
 config

Add PDB1 3rd source config.
PVT introducing multiple power sensor sources, efu-$vendor was appended
to AM4 for distinction from earlier phases.

Since the 3rd-source power monitor ISL28022 can't share the same driver
as the main and 2nd sources, we switched to dynamically loading the
driver with default resistor values. After current load testing, we
adjusted each power sensor's scale parameter based on the results.

Gerrit: https://gerrit.openbmc.org/c/openbmc/entity-manager/+/93298/2

Change-Id: Ie37493d417c051d388962bbed2f468c48cc06346
Signed-off-by: Fred Chen <Fred.Chen@quantatw.com>
---
 configurations/meson.build                    |   2 +
 .../meta/santabarbara/santabarbara_pdb1.json  |  38 +-
 .../santabarbara_pdb1_vr_adv_efu_tos.json     | 526 ++++++++++++++++++
 .../santabarbara_pdb1_vr_sni.json             |  38 +-
 .../santabarbara_pdb1_vr_sni_efu_rds.json     | 526 ++++++++++++++++++
 5 files changed, 1096 insertions(+), 34 deletions(-)
 create mode 100644 configurations/meta/santabarbara/santabarbara_pdb1_vr_adv_efu_tos.json
 create mode 100644 configurations/meta/santabarbara/santabarbara_pdb1_vr_sni_efu_rds.json

diff --git a/configurations/meson.build b/configurations/meson.build
index f3be293..af81296 100644
--- a/configurations/meson.build
+++ b/configurations/meson.build
@@ -213,7 +213,9 @@ configs = [
     'meta/santabarbara/santabarbara_pcie_switch_board_brcm.json',
     'meta/santabarbara/santabarbara_pcie_switch_board_brcm_vr_sni.json',
     'meta/santabarbara/santabarbara_pdb1.json',
+    'meta/santabarbara/santabarbara_pdb1_vr_adv_efu_tos.json',
     'meta/santabarbara/santabarbara_pdb1_vr_sni.json',
+    'meta/santabarbara/santabarbara_pdb1_vr_sni_efu_rds.json',
     'meta/santabarbara/santabarbara_pdb2.json',
     'meta/santabarbara/santabarbara_pdb2_vr_ti.json',
     'meta/santabarbara/santabarbara_pttv.json',
diff --git a/configurations/meta/santabarbara/santabarbara_pdb1.json b/configurations/meta/santabarbara/santabarbara_pdb1.json
index 09af6ca..0634a34 100644
--- a/configurations/meta/santabarbara/santabarbara_pdb1.json
+++ b/configurations/meta/santabarbara/santabarbara_pdb1.json
@@ -42,6 +42,8 @@
                 }
             ],
             "Type": "INA238",
+            "in0_Max": 1,
+            "in0_Min": -1,
             "in0_Name": "PDB1_P52V_SENSE_VOLT_V"
         },
         {
@@ -70,6 +72,8 @@
                 }
             ],
             "Type": "INA238",
+            "in0_Max": 1,
+            "in0_Min": -1,
             "in0_Name": "PDB1_GND_SENSE_VOLT_V"
         },
         {
@@ -143,12 +147,12 @@
             ],
             "Type": "INA238",
             "curr1_Name": "PDB1_P52V_ASIC0_SENSE_CURR_A",
-            "curr1_Offset": -0.0203,
-            "curr1_Scale": 994.629003382,
+            "curr1_Offset": 0.0382,
+            "curr1_Scale": 199.948013516,
             "in1_Name": "PDB1_P52V_ASIC0_SENSE_VOLT_V",
             "power1_Name": "PDB1_P52V_ASIC0_SENSE_PWR_W",
-            "power1_Offset": -1.2513,
-            "power1_Scale": 992161.920825
+            "power1_Offset": 3.4139,
+            "power1_Scale": 200803.212851406
         },
         {
             "Address": "0x44",
@@ -195,12 +199,12 @@
             ],
             "Type": "INA238",
             "curr1_Name": "PDB1_P52V_ASIC1_SENSE_CURR_A",
-            "curr1_Offset": 0.035,
-            "curr1_Scale": 1000.200040008,
+            "curr1_Offset": 0.0382,
+            "curr1_Scale": 199.948013516,
             "in1_Name": "PDB1_P52V_ASIC1_SENSE_VOLT_V",
             "power1_Name": "PDB1_P52V_ASIC1_SENSE_PWR_W",
-            "power1_Offset": 1.9937,
-            "power1_Scale": 1003814.495081
+            "power1_Offset": 3.4139,
+            "power1_Scale": 200803.212851406
         },
         {
             "Address": "0x41",
@@ -247,12 +251,12 @@
             ],
             "Type": "INA238",
             "curr1_Name": "PDB1_P52V_ASIC2_SENSE_CURR_A",
-            "curr1_Offset": 0.0355,
-            "curr1_Scale": 1001.50225338,
+            "curr1_Offset": 0.0382,
+            "curr1_Scale": 199.948013516,
             "in1_Name": "PDB1_P52V_ASIC2_SENSE_VOLT_V",
             "power1_Name": "PDB1_P52V_ASIC2_SENSE_PWR_W",
-            "power1_Offset": 2.0714,
-            "power1_Scale": 1004217.7144
+            "power1_Offset": 3.4139,
+            "power1_Scale": 200803.212851406
         },
         {
             "Address": "0x40",
@@ -299,12 +303,12 @@
             ],
             "Type": "INA238",
             "curr1_Name": "PDB1_P52V_ASIC3_SENSE_CURR_A",
-            "curr1_Offset": 0.0115,
-            "curr1_Scale": 996.909580301,
+            "curr1_Offset": 0.0382,
+            "curr1_Scale": 199.948013516,
             "in1_Name": "PDB1_P52V_ASIC3_SENSE_VOLT_V",
             "power1_Name": "PDB1_P52V_ASIC3_SENSE_PWR_W",
-            "power1_Offset": 0.0115,
-            "power1_Scale": 1000300.090027
+            "power1_Offset": 3.4139,
+            "power1_Scale": 200803.212851406
         },
         {
             "Address": "0x43",
@@ -498,7 +502,7 @@
         }
     ],
     "Name": "Santa Barbara PDB1",
-    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Santa Barbara PDB1', 'BOARD_MANUFACTURER': 'Quanta', 'BOARD_INFO_AM4': 'vr-adv'})",
+    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Santa Barbara PDB1', 'BOARD_MANUFACTURER': 'Quanta', 'BOARD_INFO_AM4': '^vr-adv$'})",
     "Type": "Board",
     "xyz.openbmc_project.Inventory.Decorator.Asset": {
         "BuildDate": "$BOARD_MANUFACTURE_DATE",
diff --git a/configurations/meta/santabarbara/santabarbara_pdb1_vr_adv_efu_tos.json b/configurations/meta/santabarbara/santabarbara_pdb1_vr_adv_efu_tos.json
new file mode 100644
index 0000000..aecfea7
--- /dev/null
+++ b/configurations/meta/santabarbara/santabarbara_pdb1_vr_adv_efu_tos.json
@@ -0,0 +1,526 @@
+{
+    "Exposes": [
+        {
+            "Address": "$address",
+            "Bus": "$bus",
+            "Name": "Santa Barbara PDB1 FRU",
+            "Type": "EEPROM"
+        },
+        {
+            "Name": "Tray_Cable_SlotID",
+            "PresencePinNames": [
+                "TRAY_SLOT_PRESENT_N"
+            ],
+            "PresencePinValues": [
+                0
+            ],
+            "Type": "GPIODeviceDetect"
+        },
+        {
+            "Address": "0x42",
+            "Bus": 45,
+            "Labels": [
+                "in1"
+            ],
+            "Name": "PDB1_P52V_SENSE",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 0.02,
+                    "Label": "in1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 1
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.02,
+                    "Label": "in1",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": -1
+                }
+            ],
+            "Type": "ISL28022",
+            "in1_Max": 1,
+            "in1_Min": -1,
+            "in1_Name": "PDB1_P52V_SENSE_VOLT_V"
+        },
+        {
+            "Address": "0x43",
+            "Bus": 45,
+            "Labels": [
+                "in1"
+            ],
+            "Name": "PDB1_GND_SENSE",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 0.02,
+                    "Label": "in1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 1
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.02,
+                    "Label": "in1",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": -1
+                }
+            ],
+            "Type": "ISL28022",
+            "in1_Max": 1,
+            "in1_Min": -1,
+            "in1_Name": "PDB1_GND_SENSE_VOLT_V"
+        },
+        {
+            "Address": "0x4c",
+            "Bus": 45,
+            "Labels": [
+                "curr1"
+            ],
+            "Name": "PDB1_BUSBAR_TSENSE_P1",
+            "Type": "ISL28022",
+            "curr1_Max": 0.1,
+            "curr1_Min": 0,
+            "curr1_Name": "PDB1_BUSBAR_TSENSE_P1_CURR_A",
+            "curr1_Scale": 50000
+        },
+        {
+            "Address": "0x4d",
+            "Bus": 45,
+            "Labels": [
+                "curr1"
+            ],
+            "Name": "PDB1_BUSBAR_TSENSE_P2",
+            "Type": "ISL28022",
+            "curr1_Max": 0.1,
+            "curr1_Min": 0,
+            "curr1_Name": "PDB1_BUSBAR_TSENSE_P2_CURR_A",
+            "curr1_Scale": 50000
+        },
+        {
+            "Address": "0x45",
+            "Bus": 46,
+            "Labels": [
+                "in0",
+                "curr1",
+                "power1"
+            ],
+            "Name": "PDB1_P52V_ASIC0_SENSE",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "in0",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "in0",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 0.7665,
+                    "Label": "curr1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 38.325
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 38,
+                    "Label": "power1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 1900.0
+                }
+            ],
+            "Type": "ISL28022",
+            "curr1_Name": "PDB1_P52V_ASIC0_SENSE_CURR_A",
+            "curr1_Offset": 0.0113,
+            "curr1_Scale": 49.48045522,
+            "in0_Name": "PDB1_P52V_ASIC0_SENSE_VOLT_V",
+            "power1_Name": "PDB1_P52V_ASIC0_SENSE_PWR_W",
+            "power1_Offset": 0.8917,
+            "power1_Scale": 50792.360828931
+        },
+        {
+            "Address": "0x44",
+            "Bus": 46,
+            "Labels": [
+                "in0",
+                "curr1",
+                "power1"
+            ],
+            "Name": "PDB1_P52V_ASIC1_SENSE",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "in0",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "in0",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 0.7665,
+                    "Label": "curr1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 38.325
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 38,
+                    "Label": "power1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 1900.0
+                }
+            ],
+            "Type": "ISL28022",
+            "curr1_Name": "PDB1_P52V_ASIC1_SENSE_CURR_A",
+            "curr1_Offset": 0.0113,
+            "curr1_Scale": 49.48045522,
+            "in0_Name": "PDB1_P52V_ASIC1_SENSE_VOLT_V",
+            "power1_Name": "PDB1_P52V_ASIC1_SENSE_PWR_W",
+            "power1_Offset": 0.8917,
+            "power1_Scale": 50792.360828931
+        },
+        {
+            "Address": "0x41",
+            "Bus": 46,
+            "Labels": [
+                "in0",
+                "curr1",
+                "power1"
+            ],
+            "Name": "PDB1_P52V_ASIC2_SENSE",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "in0",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "in0",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 0.7665,
+                    "Label": "curr1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 38.325
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 38,
+                    "Label": "power1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 1900.0
+                }
+            ],
+            "Type": "ISL28022",
+            "curr1_Name": "PDB1_P52V_ASIC2_SENSE_CURR_A",
+            "curr1_Offset": 0.0113,
+            "curr1_Scale": 49.48045522,
+            "in0_Name": "PDB1_P52V_ASIC2_SENSE_VOLT_V",
+            "power1_Name": "PDB1_P52V_ASIC2_SENSE_PWR_W",
+            "power1_Offset": 0.8917,
+            "power1_Scale": 50792.360828931
+        },
+        {
+            "Address": "0x40",
+            "Bus": 46,
+            "Labels": [
+                "in0",
+                "curr1",
+                "power1"
+            ],
+            "Name": "PDB1_P52V_ASIC3_SENSE",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "in0",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "in0",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 0.7665,
+                    "Label": "curr1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 38.325
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 38,
+                    "Label": "power1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 1900.0
+                }
+            ],
+            "Type": "ISL28022",
+            "curr1_Name": "PDB1_P52V_ASIC3_SENSE_CURR_A",
+            "curr1_Offset": 0.0113,
+            "curr1_Scale": 49.48045522,
+            "in0_Name": "PDB1_P52V_ASIC3_SENSE_VOLT_V",
+            "power1_Name": "PDB1_P52V_ASIC3_SENSE_PWR_W",
+            "power1_Offset": 0.8917,
+            "power1_Scale": 50792.360828931
+        },
+        {
+            "Address": "0x43",
+            "Bus": 47,
+            "Labels": [
+                "iout1",
+                "pin",
+                "temp1",
+                "vin",
+                "vout1"
+            ],
+            "Name": "PDB1_P52V_HSC1",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 2,
+                    "Label": "temp1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 100.0
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "vin",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "vin",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "vout1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "vout1",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 2.4,
+                    "Label": "iout1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 120
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 122.4,
+                    "Label": "pin",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 6120
+                }
+            ],
+            "Type": "LTC4287",
+            "iout1_Name": "PDB1_P52V_HSC1_CURR_A",
+            "iout1_Offset": 0.3,
+            "iout1_Scale": 1020.408163265,
+            "pin_Name": "PDB1_P52V_HSC1_PWR_W",
+            "pin_Offset": 18.8,
+            "pin_Scale": 1030927.835052,
+            "temp1_Name": "PDB1_P52V_HSC1_TEMP_C",
+            "vin_Name": "PDB1_P52V_HSC1_VIN_VOLT_V",
+            "vout1_Name": "PDB1_P52V_HSC1_VOUT_VOLT_V"
+        },
+        {
+            "Address": "0x41",
+            "Bus": 47,
+            "Labels": [
+                "iout1",
+                "pin",
+                "temp1",
+                "vin",
+                "vout1"
+            ],
+            "Name": "PDB1_P52V_HSC2",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 2,
+                    "Label": "temp1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 100.0
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "vin",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "vin",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "vout1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "vout1",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 3.4,
+                    "Label": "iout1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 170
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 173.4,
+                    "Label": "pin",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 8670
+                }
+            ],
+            "Type": "LTC4287",
+            "iout1_Name": "PDB1_P52V_HSC2_CURR_A",
+            "iout1_Offset": 0.11,
+            "iout1_Scale": 1086.956521739,
+            "pin_Name": "PDB1_P52V_HSC2_PWR_W",
+            "pin_Offset": 8.47,
+            "pin_Scale": 1098901.098901,
+            "temp1_Name": "PDB1_P52V_HSC2_TEMP_C",
+            "vin_Name": "PDB1_P52V_HSC2_VIN_VOLT_V",
+            "vout1_Name": "PDB1_P52V_HSC2_VOUT_VOLT_V"
+        },
+        {
+            "Address": "0x4a",
+            "Bus": 44,
+            "Name": "PDB1_TOP_TEMP_C",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.6,
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 80.0
+                }
+            ],
+            "Type": "TMP75"
+        },
+        {
+            "Address": "0x4e",
+            "Bus": 44,
+            "Name": "PDB1_BOT_TEMP_C",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.6,
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 80.0
+                }
+            ],
+            "Type": "TMP75"
+        }
+    ],
+    "Name": "Santa Barbara PDB1",
+    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Santa Barbara PDB1', 'BOARD_MANUFACTURER': 'Quanta', 'BOARD_INFO_AM4': '^vr-adv efu-tos$'})",
+    "Type": "Board",
+    "xyz.openbmc_project.Inventory.Decorator.Asset": {
+        "BuildDate": "$BOARD_MANUFACTURE_DATE",
+        "Manufacturer": "$BOARD_MANUFACTURER",
+        "Model": "$BOARD_PRODUCT_NAME",
+        "PartNumber": "$BOARD_PART_NUMBER",
+        "SerialNumber": "$BOARD_SERIAL_NUMBER",
+        "SparePartNumber": "$BOARD_INFO_AM1"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.AssetTag": {
+        "AssetTag": "$PRODUCT_ASSET_TAG"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.Compatible": {
+        "Names": [
+            "com.meta.Hardware.Santabarbara.Boards.Power_Distribution_Board_1"
+        ]
+    },
+    "xyz.openbmc_project.Inventory.Decorator.Revision": {
+        "Version": "$PRODUCT_VERSION"
+    }
+}
diff --git a/configurations/meta/santabarbara/santabarbara_pdb1_vr_sni.json b/configurations/meta/santabarbara/santabarbara_pdb1_vr_sni.json
index ca37cc7..aeeec67 100644
--- a/configurations/meta/santabarbara/santabarbara_pdb1_vr_sni.json
+++ b/configurations/meta/santabarbara/santabarbara_pdb1_vr_sni.json
@@ -42,6 +42,8 @@
                 }
             ],
             "Type": "INA238",
+            "in0_Max": 1,
+            "in0_Min": -1,
             "in0_Name": "PDB1_P52V_SENSE_VOLT_V"
         },
         {
@@ -70,6 +72,8 @@
                 }
             ],
             "Type": "INA238",
+            "in0_Max": 1,
+            "in0_Min": -1,
             "in0_Name": "PDB1_GND_SENSE_VOLT_V"
         },
         {
@@ -143,12 +147,12 @@
             ],
             "Type": "INA238",
             "curr1_Name": "PDB1_P52V_ASIC0_SENSE_CURR_A",
-            "curr1_Offset": -0.0203,
-            "curr1_Scale": 994.629003382,
+            "curr1_Offset": 0.0382,
+            "curr1_Scale": 199.948013516,
             "in1_Name": "PDB1_P52V_ASIC0_SENSE_VOLT_V",
             "power1_Name": "PDB1_P52V_ASIC0_SENSE_PWR_W",
-            "power1_Offset": -1.2513,
-            "power1_Scale": 992161.920825
+            "power1_Offset": 3.4139,
+            "power1_Scale": 200803.212851406
         },
         {
             "Address": "0x44",
@@ -195,12 +199,12 @@
             ],
             "Type": "INA238",
             "curr1_Name": "PDB1_P52V_ASIC1_SENSE_CURR_A",
-            "curr1_Offset": 0.035,
-            "curr1_Scale": 1000.200040008,
+            "curr1_Offset": 0.0382,
+            "curr1_Scale": 199.948013516,
             "in1_Name": "PDB1_P52V_ASIC1_SENSE_VOLT_V",
             "power1_Name": "PDB1_P52V_ASIC1_SENSE_PWR_W",
-            "power1_Offset": 1.9937,
-            "power1_Scale": 1003814.495081
+            "power1_Offset": 3.4139,
+            "power1_Scale": 200803.212851406
         },
         {
             "Address": "0x41",
@@ -247,12 +251,12 @@
             ],
             "Type": "INA238",
             "curr1_Name": "PDB1_P52V_ASIC2_SENSE_CURR_A",
-            "curr1_Offset": 0.0355,
-            "curr1_Scale": 1001.50225338,
+            "curr1_Offset": 0.0382,
+            "curr1_Scale": 199.948013516,
             "in1_Name": "PDB1_P52V_ASIC2_SENSE_VOLT_V",
             "power1_Name": "PDB1_P52V_ASIC2_SENSE_PWR_W",
-            "power1_Offset": 2.0714,
-            "power1_Scale": 1004217.7144
+            "power1_Offset": 3.4139,
+            "power1_Scale": 200803.212851406
         },
         {
             "Address": "0x40",
@@ -299,12 +303,12 @@
             ],
             "Type": "INA238",
             "curr1_Name": "PDB1_P52V_ASIC3_SENSE_CURR_A",
-            "curr1_Offset": 0.0115,
-            "curr1_Scale": 996.909580301,
+            "curr1_Offset": 0.0382,
+            "curr1_Scale": 199.948013516,
             "in1_Name": "PDB1_P52V_ASIC3_SENSE_VOLT_V",
             "power1_Name": "PDB1_P52V_ASIC3_SENSE_PWR_W",
-            "power1_Offset": 0.0115,
-            "power1_Scale": 1000300.090027
+            "power1_Offset": 3.4139,
+            "power1_Scale": 200803.212851406
         },
         {
             "Address": "0x4a",
@@ -498,7 +502,7 @@
         }
     ],
     "Name": "Santa Barbara PDB1",
-    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Santa Barbara PDB1', 'BOARD_MANUFACTURER': 'Quanta', 'BOARD_INFO_AM4': 'vr-sni'})",
+    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Santa Barbara PDB1', 'BOARD_MANUFACTURER': 'Quanta', 'BOARD_INFO_AM4': '^vr-sni$'})",
     "Type": "Board",
     "xyz.openbmc_project.Inventory.Decorator.Asset": {
         "BuildDate": "$BOARD_MANUFACTURE_DATE",
diff --git a/configurations/meta/santabarbara/santabarbara_pdb1_vr_sni_efu_rds.json b/configurations/meta/santabarbara/santabarbara_pdb1_vr_sni_efu_rds.json
new file mode 100644
index 0000000..ca33ae9
--- /dev/null
+++ b/configurations/meta/santabarbara/santabarbara_pdb1_vr_sni_efu_rds.json
@@ -0,0 +1,526 @@
+{
+    "Exposes": [
+        {
+            "Address": "$address",
+            "Bus": "$bus",
+            "Name": "Santa Barbara PDB1 FRU",
+            "Type": "EEPROM"
+        },
+        {
+            "Name": "Tray_Cable_SlotID",
+            "PresencePinNames": [
+                "TRAY_SLOT_PRESENT_N"
+            ],
+            "PresencePinValues": [
+                0
+            ],
+            "Type": "GPIODeviceDetect"
+        },
+        {
+            "Address": "0x42",
+            "Bus": 45,
+            "Labels": [
+                "in0"
+            ],
+            "Name": "PDB1_P52V_SENSE",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 0.02,
+                    "Label": "in0",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 1
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.02,
+                    "Label": "in0",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": -1
+                }
+            ],
+            "Type": "SQ52206",
+            "in0_Max": 1,
+            "in0_Min": -1,
+            "in0_Name": "PDB1_P52V_SENSE_VOLT_V"
+        },
+        {
+            "Address": "0x43",
+            "Bus": 45,
+            "Labels": [
+                "in0"
+            ],
+            "Name": "PDB1_GND_SENSE",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 0.02,
+                    "Label": "in0",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 1
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.02,
+                    "Label": "in0",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": -1
+                }
+            ],
+            "Type": "SQ52206",
+            "in0_Max": 1,
+            "in0_Min": -1,
+            "in0_Name": "PDB1_GND_SENSE_VOLT_V"
+        },
+        {
+            "Address": "0x4c",
+            "Bus": 45,
+            "Labels": [
+                "curr1"
+            ],
+            "Name": "PDB1_BUSBAR_TSENSE_P1",
+            "Type": "SQ52206",
+            "curr1_Max": 0.1,
+            "curr1_Min": 0,
+            "curr1_Name": "PDB1_BUSBAR_TSENSE_P1_CURR_A",
+            "curr1_Scale": 200000
+        },
+        {
+            "Address": "0x4d",
+            "Bus": 45,
+            "Labels": [
+                "curr1"
+            ],
+            "Name": "PDB1_BUSBAR_TSENSE_P2",
+            "Type": "SQ52206",
+            "curr1_Max": 0.1,
+            "curr1_Min": 0,
+            "curr1_Name": "PDB1_BUSBAR_TSENSE_P2_CURR_A",
+            "curr1_Scale": 200000
+        },
+        {
+            "Address": "0x45",
+            "Bus": 46,
+            "Labels": [
+                "in1",
+                "curr1",
+                "power1"
+            ],
+            "Name": "PDB1_P52V_ASIC0_SENSE",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "in1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "in1",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 0.7665,
+                    "Label": "curr1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 38.325
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 38,
+                    "Label": "power1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 1900.0
+                }
+            ],
+            "Type": "SQ52206",
+            "curr1_Name": "PDB1_P52V_ASIC0_SENSE_CURR_A",
+            "curr1_Offset": 0.0382,
+            "curr1_Scale": 199.948013516,
+            "in1_Name": "PDB1_P52V_ASIC0_SENSE_VOLT_V",
+            "power1_Name": "PDB1_P52V_ASIC0_SENSE_PWR_W",
+            "power1_Offset": 3.4139,
+            "power1_Scale": 200803.212851406
+        },
+        {
+            "Address": "0x44",
+            "Bus": 46,
+            "Labels": [
+                "in1",
+                "curr1",
+                "power1"
+            ],
+            "Name": "PDB1_P52V_ASIC1_SENSE",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "in1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "in1",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 0.7665,
+                    "Label": "curr1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 38.325
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 38,
+                    "Label": "power1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 1900.0
+                }
+            ],
+            "Type": "SQ52206",
+            "curr1_Name": "PDB1_P52V_ASIC1_SENSE_CURR_A",
+            "curr1_Offset": 0.0382,
+            "curr1_Scale": 199.948013516,
+            "in1_Name": "PDB1_P52V_ASIC1_SENSE_VOLT_V",
+            "power1_Name": "PDB1_P52V_ASIC1_SENSE_PWR_W",
+            "power1_Offset": 3.4139,
+            "power1_Scale": 200803.212851406
+        },
+        {
+            "Address": "0x41",
+            "Bus": 46,
+            "Labels": [
+                "in1",
+                "curr1",
+                "power1"
+            ],
+            "Name": "PDB1_P52V_ASIC2_SENSE",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "in1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "in1",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 0.7665,
+                    "Label": "curr1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 38.325
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 38,
+                    "Label": "power1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 1900.0
+                }
+            ],
+            "Type": "SQ52206",
+            "curr1_Name": "PDB1_P52V_ASIC2_SENSE_CURR_A",
+            "curr1_Offset": 0.0382,
+            "curr1_Scale": 199.948013516,
+            "in1_Name": "PDB1_P52V_ASIC2_SENSE_VOLT_V",
+            "power1_Name": "PDB1_P52V_ASIC2_SENSE_PWR_W",
+            "power1_Offset": 3.4139,
+            "power1_Scale": 200803.212851406
+        },
+        {
+            "Address": "0x40",
+            "Bus": 46,
+            "Labels": [
+                "in1",
+                "curr1",
+                "power1"
+            ],
+            "Name": "PDB1_P52V_ASIC3_SENSE",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "in1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "in1",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 0.7665,
+                    "Label": "curr1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 38.325
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 38,
+                    "Label": "power1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 1900.0
+                }
+            ],
+            "Type": "SQ52206",
+            "curr1_Name": "PDB1_P52V_ASIC3_SENSE_CURR_A",
+            "curr1_Offset": 0.0382,
+            "curr1_Scale": 199.948013516,
+            "in1_Name": "PDB1_P52V_ASIC3_SENSE_VOLT_V",
+            "power1_Name": "PDB1_P52V_ASIC3_SENSE_PWR_W",
+            "power1_Offset": 3.4139,
+            "power1_Scale": 200803.212851406
+        },
+        {
+            "Address": "0x4a",
+            "Bus": 44,
+            "Name": "PDB1_TOP_TEMP_C",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.6,
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 80.0
+                }
+            ],
+            "Type": "TMP75"
+        },
+        {
+            "Address": "0x4e",
+            "Bus": 44,
+            "Name": "PDB1_BOT_TEMP_C",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.6,
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 80.0
+                }
+            ],
+            "Type": "TMP75"
+        },
+        {
+            "Address": "0x14",
+            "Bus": 47,
+            "Labels": [
+                "iout1",
+                "pin",
+                "temp1",
+                "vin",
+                "vout1"
+            ],
+            "Name": "PDB1_P52V_HSC1",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 2,
+                    "Label": "temp1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 100.0
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "vin",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "vin",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "vout1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "vout1",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 2.4,
+                    "Label": "iout1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 120
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 122.4,
+                    "Label": "pin",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 6120
+                }
+            ],
+            "Type": "XDP710",
+            "iout1_Name": "PDB1_P52V_HSC1_CURR_A",
+            "iout1_Offset": 0.14,
+            "iout1_Scale": 806.451612903,
+            "pin_Name": "PDB1_P52V_HSC1_PWR_W",
+            "pin_Offset": 6,
+            "pin_Scale": 806451.612903226,
+            "temp1_Name": "PDB1_P52V_HSC1_TEMP_C",
+            "vin_Name": "PDB1_P52V_HSC1_VIN_VOLT_V",
+            "vout1_Name": "PDB1_P52V_HSC1_VOUT_VOLT_V"
+        },
+        {
+            "Address": "0x16",
+            "Bus": 47,
+            "Labels": [
+                "iout1",
+                "pin",
+                "temp1",
+                "vin",
+                "vout1"
+            ],
+            "Name": "PDB1_P52V_HSC2",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 2,
+                    "Label": "temp1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 100.0
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "vin",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "vin",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 1.12,
+                    "Label": "vout1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 56
+                },
+                {
+                    "Direction": "less than",
+                    "Hysteresis": 0.92,
+                    "Label": "vout1",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 46
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 3.4,
+                    "Label": "iout1",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 170
+                },
+                {
+                    "Direction": "greater than",
+                    "Hysteresis": 173.4,
+                    "Label": "pin",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 8670
+                }
+            ],
+            "Type": "XDP710",
+            "iout1_Name": "PDB1_P52V_HSC2_CURR_A",
+            "iout1_Offset": -0.12,
+            "iout1_Scale": 510.204081633,
+            "pin_Name": "PDB1_P52V_HSC2_PWR_W",
+            "pin_Offset": -6,
+            "pin_Scale": 512820.512820513,
+            "temp1_Name": "PDB1_P52V_HSC2_TEMP_C",
+            "vin_Name": "PDB1_P52V_HSC2_VIN_VOLT_V",
+            "vout1_Name": "PDB1_P52V_HSC2_VOUT_VOLT_V"
+        }
+    ],
+    "Name": "Santa Barbara PDB1",
+    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Santa Barbara PDB1', 'BOARD_MANUFACTURER': 'Quanta', 'BOARD_INFO_AM4': '^vr-sni efu-rds$'})",
+    "Type": "Board",
+    "xyz.openbmc_project.Inventory.Decorator.Asset": {
+        "BuildDate": "$BOARD_MANUFACTURE_DATE",
+        "Manufacturer": "$BOARD_MANUFACTURER",
+        "Model": "$BOARD_PRODUCT_NAME",
+        "PartNumber": "$BOARD_PART_NUMBER",
+        "SerialNumber": "$BOARD_SERIAL_NUMBER",
+        "SparePartNumber": "$BOARD_INFO_AM1"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.AssetTag": {
+        "AssetTag": "$PRODUCT_ASSET_TAG"
+    },
+    "xyz.openbmc_project.Inventory.Decorator.Compatible": {
+        "Names": [
+            "com.meta.Hardware.Santabarbara.Boards.Power_Distribution_Board_1"
+        ]
+    },
+    "xyz.openbmc_project.Inventory.Decorator.Revision": {
+        "Version": "$PRODUCT_VERSION"
+    }
+}
-- 
2.39.3

